### PR TITLE
(itests, test-framework) Handle "Password Hint needed" shell when it pops up.

### PR DESCRIPTION
**Pull request by rhopp**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by bdshadow** [comment]
testPR

**Comment by rhopp** [comment]
I've added log into the empty catch block (as sonarqube suggested). If it is ok, please merge.

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/OpenShiftLabel.java [hunk 1]
```diff
@@ -176,6 +176,7 @@
 		public static final String RESTORE_SNAPSHOT = "Restore/Deploy Snapshot";
 		public static final String SECURE_STORAGE = "Secure Storage";
 		public static final String SECURE_STORAGE_PASSWORD = "Secure Storage Password";
```
Comment (adietish) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/OpenShiftLabel.java]: false negative, this isnt a password

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/OpenShiftLabel.java [hunk 2]
```diff
@@ -176,6 +176,7 @@
...
+		public static final String PASSWORD_HINT_NEEDED = "Secure Storage - Password Hint Needed";
```
Comment (adietish) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/OpenShiftLabel.java]: false negative, this isnt a password

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java [hunk 3]
```diff
@@ -35,6 +38,8 @@
 
 public class SecureStorage {
```
Comment (adietish) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java]: +1, correctly identified, @rhopp can you please apply this simple suggestion?

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java [hunk 4]
```diff
@@ -86,8 +87,15 @@ private static void triggerSecureStorageOfPasswordInConnectionDialog(String user
 			boolean firstStorage = provideSecureStoragePassword(SystemProperties.SECURE_STORAGE_PASSWORD);
 			
 			if (firstStorage) {
-				new DefaultShell(OpenShiftLabel.Shell.SECURE_STORAGE).setFocus(); 
-				new NoButton().click();	
+				// Did "Password hint needed" shell appear? Get rid of it.
+				try {
+					new DefaultShell(OpenShiftLabel.Shell.PASSWORD_HINT_NEEDED);
+					new NoButton().click();
+				} catch (SWTLayerException ex) {
```
Comment (adietish) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java]: false negative, the exception is being logged. I dont see any flaw here

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java (hunk 3 above)
Comment (rhopp) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java]: Sure. Thx for review.

Comment (rhopp) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/SecureStorage.java]: Private constructor added.

**Pull request merged**